zerver/lib/upload/local.py

```diff
@@ -103,9 +103,12 @@ def delete_message_attachment(self, path_id: str) -> bool:
         return delete_local_file("files", path_id)
 
     @override
-    def all_message_attachments(self) -> Iterator[Tuple[str, datetime]]:
+    def all_message_attachments(self, thumbnails: bool = False) -> Iterator[Tuple[str, datetime]]:
         assert settings.LOCAL_UPLOADS_DIR is not None
-        for dirname, _, files in os.walk(settings.LOCAL_UPLOADS_DIR + "/files"):
+        top = settings.LOCAL_UPLOADS_DIR + "/files"
+        for dirname, subdirnames, files in os.walk(top):
+            if not thumbnails and dirname == top and "thumbnail" in subdirnames:
+                subdirnames.remove("thumbnail")
```

Comment:
Would `include_thumbnails` be a better name here? It seems like it includes the non-thumbnails unconditionally.